### PR TITLE
wallet redesign

### LINKED ISSUE #2174: Feature: wallet redesign
- [x] assess coin gecko api
https://www.coingecko.com/en/api/documentation

- [x] package for chart
- [x] assess/sandbox test api calls involved in wallet
- [x] wallet main screen components
- [x] wallet details screen components
- [x] make filters functional
- [x] integrate wallet actions
- [x] add missing balances (if HIVE or HBD is pending conversion or reserved in market open orders, etc.) We should still show them and inform people about their other/pending/locked balances. On website, we recently added those, HBD if being converted, it takes 3 days to convert and HBD locked during that time. If HIVE or HBD sell/buy order created (not full-filled by market orders yet, then It is locked as well) When HBD/HIVE being withdrawn from Savings, it takes 3.5 days also locked, etc. etc.

### PR DESCRIPTION
**Pull request by noumantahir**
### What does this PR?
Very early stage wallet redesing.

In Progress: wallet currency card

### Issue number
fixes #2174

### Screenshots/Video
Screenshot 2022-02-05 at 11 05 49 PM

### CODE REVIEW

**Comment by noumantahir** [comment]
fine tuned currency card....

Screenshot 2022-02-07 at 12 52 10 AM
Screenshot 2022-02-07 at 12 49 53 AM

**Comment by noumantahir** [comment]
Update:
Loaded main screen with real data, rearranged  API calls, optimised data, fined tuned UI a bit.

Screenshot 2022-02-12 at 12 21 38 AM
Screenshot 2022-02-12 at 12 21 59 AM

**Comment by feruzm** [comment]
@noumantahir i would vote for rewrite because we will be doing different things and of course you can copy paste what you need (e.g. functions) from old container or from website to boost dev time.

**Comment by noumantahir** [comment]
Updates:

coin card list now integrates with real time coin basic data through redux, moving forward we can reuse the same object more more coin/token types we add support for.


Screenshot 2022-02-20 at 9 22 20 AM
Screenshot 2022-02-20 at 9 44 35 AM

**Comment by noumantahir** [comment]
@feruzm what do you think of this deviation for claiming and getting more rewards.

Screenshot 2022-02-20 at 9 37 30 AM
Screenshot 2022-02-20 at 9 22 20 AM

File: src/utils/wallet.ts [hunk 1]
```diff
@@ -242,6 +252,159 @@ export const groomingWalletData = async (user, globalProps, userCurrency) => {
...
+      return walletData;
+    }
+
+    //TODO: Use already available accoutn for frist wallet start
+    const state = await getAccount(username);
+    const ecencyUser = await getEcencyUser(username)
+    //TODO: cache data in redux or fetch once on wallet startup
+    const ppHbd = await getCurrencyTokenRate(vsCurrency, 'hbd');
+    const ppHive = await getCurrencyTokenRate(vsCurrency, 'hive');
+    const ppEstm = await getCurrencyTokenRate(vsCurrency, 'estm');
```
Comment (noumantahir) [inline src/utils/wallet.ts:280]: @feruzm this method returns estm token value to be 0.002, however, I remember equating 1 point to 1/150 usd while the returned value equates to 1/500... which should I use for Ecency points calculation...?

Screenshot 2022-02-20 at 9 49 21 AM

Comment (feruzm) [reply src/utils/wallet.ts:280]: @noumantahir there is purchase value and there is content boosting value, they are different.

**Comment by feruzm** [comment]
@noumantahir as for claiming buttons, can we use same button without ‘claim’ word in it but have + icon, for consistency across the apps and website. Other than that, looks good

**Comment by noumantahir** [comment]
@feruzm need your feedback on this, HP had a number to data to be shown in header... serialised extra data in a different format to make primary data look prominent... approach can be extended for other tokens flexibly as well if need be in future tokens/coins.

Screenshot 2022-02-21 at 4 22 58 PM

**Comment by feruzm** [comment]
@noumantahir i like picture on right, extra info can be smaller

**Comment by feruzm** [comment]
@noumantahir i like it a lot, operations are missing so far and probably history filters right?!

**Comment by noumantahir** [comment]
wallet branch is now being built in app centre, beta should be available as build is complete

**Comment by noumantahir** [comment]
> @noumantahir i like it a lot, operations are missing so far and probably history filters right?!

strictly a work in progress.. a looot is missing

**Comment by noumantahir** [comment]
@feruzm you can track and add/suggest pending items here. #2174

**Comment by noumantahir** [comment]
Massive improvements in first time wallet load by used pre-cached data....

https://user-images.githubusercontent.com/6298342/155598176-71afaf22-1db4-4101-9a92-9b97a452e170.mov

**Comment by noumantahir** [comment]
Update: added support for currency and user change, integrated translations

**Comment by noumantahir** [comment]
@feruzm Updates:

- Added support for coin data update on transfer complete
- Two row balance representation
- Improved unclaimed balance string

**Comment by noumantahir** [comment]
@feruzm spent some time testing wallet, we should be able to merge now... test well and let me know your thoughts

**Comment by feruzm** [comment]
@noumantahir great, will do code review now 😉

File: src/config/locales/en-US.json [hunk 2]
```diff
@@ -93,20 +100,35 @@
...
     },
     "btc": {
       "title": "BTC",
       "buy": "GET BTC",
       "address": "RECEIVE"
-    }
+    },
+    "last_updated":"Last Updated:",
+    "updating":"Updating...",
+    "coin_details":"Coin Details",
```
Comment (feruzm) [inline src/config/locales/en-US.json:125]: token, coin, too much different terms imo, maybe just Details would suffice?

File: src/config/locales/en-US.json [hunk 3]
```diff
@@ -93,20 +100,35 @@
...
+    "change":"Change",
+    "activities":"Activities",
+    "savings_withdrawal":"Pending Withdrawals",
+    "open_orders":"Open Orders",
+    "conversions_requested":"Converstions Requested",
```
Comment (feruzm) [inline src/config/locales/en-US.json:130]: typo here

File: src/screens/coinDetails/children/activitiesList.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  const _renderActivityItem = ({ item, index }) => {
+    return <Transaction item={item} index={index} />
+  }
+
+  const sections = [];
+
+  if (pendingActivities && pendingActivities.length) {
+    sections.push({
+      title: 'Pending Requests',
```
Comment (feruzm) [inline src/screens/coinDetails/children/activitiesList.tsx:27]: String should be in locales...

File: src/screens/wallet/screen/walletScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+      Alert.alert(`Failed to claim rewards, ${error.message}\nTry again or write to support@ecency.com`);
+    }
+    setIsClaiming(false);
+  }
+
+
+  const _claimRewards = (coinId:string) => {
+    if(isLoading){
+      setRefreshing(true);
+      Alert.alert("Wallet update in progress, try again as update finishes");
```
Comment (feruzm) [inline src/screens/wallet/screen/walletScreen.tsx:154]: should be in translation locales

**Review by feruzm — CHANGES_REQUESTED** [review]

**Review by feruzm — APPROVED** [review]
🎉

**Pull request merged**